my-code/movie/src/App.js

```diff
@@ -0,0 +1,74 @@
+
+class App extends Component {
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      search: "",
+      detail: "",
+      fav: [],
+      results: movies.Search//false
+    }
+
+    this.changeText = this.changeText.bind(this)
+    this.getMovies = this.getMovies.bind(this)
+    this.getDetails = this.getDetails.bind(this)
+    this.handleLikes = this.handleLikes.bind(this)
+  }
+
+  handleLikes(id) {    
+    const remove = (filmID) => {this.state.fav.filter(fav => {
+      return fav !== filmID
+    })}
+
+    if(this.state.fav.includes(id)){
+      let newFav = remove(id)
+      return this.setState({fav: newFav})
+    }
+    this.setState({fav: [id, ...this.state.fav]})
+  }
+
+  changeText(event) {
+    if(event.key === 'Enter')
+      return this.getMovies
+    this.setState({
+      search: event.target.value
+    }, this.getMovies)
+  }
+
+  async getDetails(id) {
+    let detail = await movieDetails(id)
+    this.setState({detail})
+    console.log(detail)
+  }
+
+  async getMovies() { 
+    if(this.state.search.length > 3) {
+      let search = await movieSearch(this.state.search)
+      this.setState({results: search.Search})
+    }
+  }
+
+  render() {
+    return (
+      <Router>
+        <div className="App">        
+          <Route exact path = '/' render={(props) => <Home {...props} res={this.state.results} search={this.state.value} searchHandler={this.changeText} submit={this.getMovies}/>} />
+          <Route path = '/details/:id' render={(props) => <Details {...props} fetch={this.getDetails} detail={this.state.detail} like={this.handleLikes}/>} />
+        </div>
+      </Router>
+      
+    );
+  }
+}
+
+export default App
```

Comment:
I think you have to choose a way to manage your data. Currently, there are a lot of options, such as Redux, React Context, MobX, etc. At here, in Significa, we prefer to use Redux because we can control the side effect, it has a single data store, it's immutable, plus there are too many tools that help our work. 

So, think about these options. It will help your job a lot  :)
Take a look at EggHead and Udemy, there are some amazing courses there!

Reply:
I know about Redux, never used it before because i never felt the need. I build small apps and i guess Redux could be an overkill. I've been wanting to dive in for Redux for quite some time...